src/main/kotlin/io/emeraldpay/dshackle/reader/SpanReader.kt

```diff
@@ -0,0 +1,57 @@
+package io.emeraldpay.dshackle.reader
+
+import io.emeraldpay.dshackle.commons.SPAN_ERROR
+import io.emeraldpay.dshackle.commons.SPAN_READER_NAME
+import io.emeraldpay.dshackle.commons.SPAN_READER_RESULT
+import io.emeraldpay.dshackle.commons.SPAN_REQUEST_INFO
+import io.emeraldpay.dshackle.commons.SPAN_STATUS_MESSAGE
+import io.emeraldpay.dshackle.data.HashId
+import io.emeraldpay.dshackle.upstream.rpcclient.JsonRpcRequest
+import org.springframework.cloud.sleuth.Tracer
+import org.springframework.cloud.sleuth.instrument.reactor.ReactorSleuth
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+class SpanReader<K, D>(
```

Comment:
I don't lile this name, looks like it is a reader of spans. It can be confusing. Maybe something like observable reader?